modules/renter/contractor/renew.go

```diff
@@ -57,133 +57,46 @@ func (c *Contractor) managedRenew(contract modules.RenterContract, numSectors ui
-		c.renewing[id] = true
+		return
 	}
-	c.mu.Unlock()
 
-	// after we finish renewing, unset the 'renewing' flag on each contract
-	defer func() {
-		c.mu.Lock()
-		for _, id := range renewSet {
-			delete(c.renewing, id)
-		}
-		c.mu.Unlock()
-	}()
-
-	// wait for all active editors and downloaders to finish, then grab the
-	// latest revision of each contract
-	var oldContracts []modules.RenterContract
-	for _, id := range renewSet {
-		c.mu.RLock()
-		e, eok := c.editors[id]
-		d, dok := c.downloaders[id]
-		c.mu.RUnlock()
-		if eok {
-			e.invalidate()
-		}
-		if dok {
-			d.invalidate()
-		}
-
-		c.mu.RLock()
-		contract, ok := c.contracts[id]
-		c.mu.RUnlock()
-		if !ok {
-			c.log.Printf("WARN: no record of contract previously added to the renew set (ID: %v)", id)
-			continue
-		}
-		oldContracts = append(oldContracts, contract)
-	}
-
-	// map old ID to new contract, for easy replacement later
-	newContracts := make(map[types.FileContractID]modules.RenterContract)
-	for _, contract := range oldContracts {
-		newContract, err := c.managedRenew(contract, numSectors, endHeight)
-		if err != nil {
-			c.log.Printf("WARN: failed to renew contract with %v: %v", contract.NetAddress, err)
-		} else {
-			newContracts[contract.ID] = newContract
-			c.log.Printf("renewed contract %v -> %v", contract.ID, newContract.ID)
-		}
-		// sleep between renewing each contract to alleviate potential block
-		// propagation issues
-		time.Sleep(contractFormationInterval)
-	}
-
-	// replace old contracts with renewed ones
+	// Success, update the set of contracts in the contractor.
+	c.log.Printf("Renewed a contract with %v for %v SC\n", host.PublicKey.String(), contract.TotalCost.Div(types.SiacoinPrecision))
 	c.mu.Lock()
-	for oldID, contract := range newContracts {
-		// archive the old contract
-		if oldContract, ok := c.contracts[oldID]; ok {
-			c.oldContracts[oldID] = oldContract
-			delete(c.contracts, oldID)
-		}
-		// insert the new contract
-		c.contracts[contract.ID] = contract
-		// add a mapping from old->new contract
-		c.renewedIDs[oldID] = contract.ID
-		// move the cachedRevision entry to the new ID
-		c.cachedRevisions[contract.ID] = c.cachedRevisions[oldID]
-		delete(c.cachedRevisions, oldID)
-	}
+	defer c.mu.Unlock()
+
+	// Establish the new contract as in good standing, copy the upload
+	// usefulness from the old contract.
+	newContract.InGoodStanding = true
+	newContract.UsefulForUpload = contract.UsefulForUpload
+	// Archive the old contract.
+	c.oldContracts[contract.ID] = contract
+	// Delete the old contract.
+	delete(c.contracts, contract.ID)
+	// Insert the new contract.
+	c.contracts[newContract.ID] = newContract
+	// Map from the old contract to the new contract.
+	c.renewedIDs[contract.ID] = newContract.ID
+	// Transfer the current cached revision to the new contract id.
+	c.cachedRevisions[newContract.ID] = c.cachedRevisions[contract.ID] // TODO: Is this necessary, won't the revision numbers, etc. be off anyway?
```

Comment:
yeah, that does seem unnecessary. It was explicitly added though (see https://github.com/NebulousLabs/Sia/commit/523f0b2c22bcfa24b1e9006e9c358a76b3c25cfc), so I must have thought it was necessary for some reason. Better commit messages would have helped here :/
Check if removing it breaks anything, I guess.